server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -520,6 +577,12 @@ export class MongoDbFactory implements core.IDbFactory {
 			global ? this.globalDbEndpoint : this.operationsDbEndpoint,
 			options,
 		);
+		for (const monitoringEvent of this.dbMonitoringEventsList) {
+			connection.on(monitoringEvent, (event) => {
+				Lumberjack.info(`Mongo DB event received: ${monitoringEvent}`, event);
```

Comment:
What if the event describes an error; should this be Lumberjack.error() instead?

One approach could be to break down the dbMonitoringEventsList in different categories, similar to log level: debug, info, error. For each one of them, upon receiving the event, we would use the appropriate Lumberjack level for that category.

Also, do we plan to alert on some of these errors? If yes, do we want to use metrics instead for some of the events?

Reply:
I like the suggestion for alerting. No plans yet but could do. However these are event listeners, meaning they are listening to some events emitted by the sdk/server. There are no really metric around this, what the best metrics usage you can think? An empty metric with a message?

As far as level based, I think it is a little bit over kill now, the events are not so many after selection:
![image](https://github.com/microsoft/FluidFramework/assets/3719246/d22a3373-850f-4a8e-b19a-7fef233eeccd)

Reply:
Understood - you are right, it's hard to model a Lumberjack metric around these events. One approach that is something like a workaround is to create a metric and immediately complete it in the event listener. We could use the same Lumberjack EventName for all of these events, telling them apart via a new dimension that could be called "MongoSDKEventName" or something similar. But maybe that is an overkill for now and not needed.

But on the other hand, I think there is value in using the appropriate log level depending on the Mongo event. At least, it doesn't seem to make sense to me to have a Lumberjack.info() if the event is an error like serverHeartbeatFailed.

Reply:
I used an immediate terminated empty event for the geneva monitoring.

However, I still think we should not classify the level here. With the metric, it seems the level specified event indicates the event recording is wrong, but the event itself is informative and nothing wrong with it.
For example, it could be some good example like "Info: Event recorded for serverHeartbeatSucceeded", but the following would seems weird:
1. "Info: Event recorded for serverHeartbeatStarted", again this is an informative event, should we use `.sucess`?
2. "Error: Event recorded for serverHeartbeatFailed failed". Same, is the error for recording the event? Is this error for any meaningful operation? No for both and it just add the level of confusion and complexity.
3. what to classify `connectionPoolCleared`? Is it good(Info) or bad(warning/error)?  It is an even result of db closing, could be both good and bad.

Therefore, I don't really think we need the level classify to complex the design for now, we don't have any predictable usage yet and these event are just informative, there is nothing wrong with these event themself.

Reply:
If we are always marking the metric as success, then what is the value of that metric? It cannot be used to track duration or latency since, as we discussed, we are sort of using it as a "hack" around the Mongo events. Also, since it is always success, then there is no way to create a monitor to track that metric.

If it does not make sense to change the telemetry metric/log level, then I would recommend reverting this back to a log.

To address your other point, I think we can avoid those confusions by also reverting back to simple log statements like
`Lumberjack.info()` and `Lumberjack.error()`.
1. For Mongo events that are clearly errors, use Lumberjack.error(). I see at least 2 in the list: `serverHeartbeatFailed`, `connectionCheckOutFailed`.
2. For others that indicate success or we are not sure how to label (such as `connectionPoolCleared` in your example above), use `Lumberjack.info()`. If we learn, later, that those represent errors, we can move them to the appropriate "level"
3. For the verbose events, you could also do `Lumberjack.verbose()` if you think that is helpful. We recently introduced log level filtering capabilities in FRS.

I think the above strategy should help us categorize those events better - in fact, that is going to be super helpful for a feature I was discussing with @manishgargd and @PingZhu2232 on how to proactively alert/monitor in MongoDB issues.

Reply:
I appreciate the idea you brought up for the metric, and I do think it will be helpful for monitoring, either using geneva metric or using icm automation logs. Using metric would benefit for both but using logs can only be useful for icm automation and subject to a lot of long chained dependencies and affected by a a lot of reliability. There are multiple options here:
1. log + level, this definitely cleaner, however it lack of the potential for the geneva monitor.
2. Just a hack event here to support geneva. To answer your question regarding how to monitor, you can see line 587: `MonitoringEventName`, and assume it can be valid dimension value(Correct me if I am wrong). We can use geneva's string filter, inclusion/exclusion for `MonitoringEventName` to create meaningful alerts as well, and it actually gives us more flexibility that we can build different monitors based on value of `MonitoringEventName`.
3. metric + level, we both agree this combination is a very confusing approach.

With all this being said, I think the pro+cron is 2 > 1 >> 3, that is why it lead me to this decision. Does this make sense to you?

Reply:
Sounds great, thanks for explaining, Xin! We can try approach 2, I think it's worth a try. Worst case scenario, we just come back here and try a different strategy, which should not be something hard to do. Thanks for working on this!